src/core/numerics/ampere/ampere.h

```diff
@@ -248,33 +179,23 @@ namespace core
                 }
             }
         }
-    };
-
 
 
 
-    template<typename GridLayout>
-    class Ampere
-    {
-    private:
-        AmpereImpl<GridLayout, GridLayout::dimension> impl_;
 
     public:
         template<typename VecField>
         void operator()(VecField const& B, VecField& J)
         {
-            if (!impl_.hasLayoutSet())
+            if (!this->hasLayout())
             {
                 throw std::runtime_error(
                     "Error - Ampere - GridLayout not set, cannot proceed to calculate ampere()");
             }
             std::cout << "I'm solving ampere " << GridLayout::dimension << "\n";
 
-            impl_(B, J);
+            this->operator()<VecField, GridLayout::dimension>(B, J);
```

Comment:
cosmetic:

Vecfield here is inferrable from B/J if the template definitions are swapped around, like

```
template<std::size_t dim, typename VecField, std::enable_if_t<dim == 1, int> = 0>`
```